### PR TITLE
Split CnToDnRequestType to sync and async & Add check for adding new request type

### PR DESCRIPTION
**Pull request by liyuheng55555**
image

### CODE REVIEW

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/client/async/CnToDnInternalServiceAsyncRequestManager.java [hunk 1]
```diff
@@ -93,299 +92,314 @@
...
   @Override
-  protected AsyncRequestRPCHandler<?, CnToDnRequestType, TDataNodeLocation> buildHandler(
-      AsyncRequestContext<?, ?, CnToDnRequestType, TDataNodeLocation> requestContext,
+  protected void checkActionMapCompleteness() {
+    List<CnToDnAsyncRequestType> lackList =
+        Arrays.stream(CnToDnAsyncRequestType.values())
+            .filter(type -> !actionMap.containsKey(type))
+            .collect(Collectors.toList());
+    if (!lackList.isEmpty()) {
+      LOGGER.error("These request types should be added to actionMap: {}", lackList);
```
[the remark this answers is not in the document]
Comment (liyuheng55555) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/client/async/CnToDnInternalServiceAsyncRequestManager.java]: Throw an UncheckedStartupException

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/client/sync/SyncDataNodeClientPool.java [hunk 2]
```diff
@@ -58,20 +62,97 @@ public class SyncDataNodeClientPool {
...
+    actionMap = actionMapBuilder.build();
+  }
+
+  private void checkActionMapCompleteness() {
+    List<CnToDnSyncRequestType> lackList =
+        Arrays.stream(CnToDnSyncRequestType.values())
+            .filter(type -> !actionMap.containsKey(type))
+            .collect(Collectors.toList());
+    if (!lackList.isEmpty()) {
+      LOGGER.error("These request types should be added to actionMap: {}", lackList);
```
[the remark this answers is not in the document]
Comment (liyuheng55555) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/client/sync/SyncDataNodeClientPool.java]: same as above

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeStartupCheck.java [hunk 3]
```diff
@@ -222,4 +225,10 @@ private void createDirIfEmpty(File dir) throws IOException {
       }
     }
   }
+
+  // The checks are in the initialization process of the RequestManager object.
+  private void checkInternalRpcService() {
```
[the remark this answers is not in the document]
Comment (liyuheng55555) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeStartupCheck.java:230]: checkCnToDnRequestManager?

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**